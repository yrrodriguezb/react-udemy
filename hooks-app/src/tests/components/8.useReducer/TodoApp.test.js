import React from 'react';
import { act } from '@testing-library/react'
import '@testing-library/jest-dom';
import { shallow, mount } from 'enzyme';
import { TodoApp } from '../../../components/_8.useRecuder/TodoApp';
import { demoTodos } from '../../fixtures/demoTodos';


describe('Pruebas en <TodoApp />', () => {
  const wrapper = shallow(<TodoApp />);

  Storage.prototype.setItem = jest.fn(() => {});


  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe existir un elemento <TodoList /> y un <TodoAddForm />', () => {
    expect(wrapper.find('TodoList').exists()).toBe(true);
    expect(wrapper.find('TodoAddForm').exists()).toBe(true);
  });

  test('Debe agregar un TODO', () => {
    const wrapper = mount(<TodoApp />);

    act(() => {
      for (let index = 0; index < demoTodos.length; index++) {
        const todo = demoTodos[index];
        wrapper.find('TodoAddForm').prop('handleAddTodo')( todo );
      }
    });

    expect(wrapper.find('h1').text(`TodoApp ${demoTodos.length}`));

    // Valida que el localStorage se llame
    expect(localStorage.setItem).toHaveBeenCalledTimes(demoTodos.length);
  });

  test('Debe eliminar un TODO', () => {
    wrapper.find('TodoAddForm').prop('handleAddTodo')( demoTodos[0] );
    wrapper.find('TodoList').prop('handleDelete')( demoTodos[0].id );

    expect(wrapper.find('h1').text(`TodoApp 0`));
  });

});