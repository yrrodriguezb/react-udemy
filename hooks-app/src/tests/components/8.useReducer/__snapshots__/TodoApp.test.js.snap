// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoApp /> Debe mostrarse correctamente 1`] = `
<div
  className="p-5"
>
  <h1>
    TodoApp 
    0
     
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <TodoList
        handleDelete={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col-5"
    >
      <TodoAddForm
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
